traits/ctraits.c

```diff
@@ -2908,22 +2897,31 @@ static trait_setattr setattr_handlers[] = {
     /*  End of __setstate__ method entries */
     NULL};
 
-static int
-trait_init(trait_object *trait, PyObject *args, PyObject *kwds)
+
+trait_object *
+trait_new (PyTypeObject *trait_type, PyObject *args, PyObject *kw)
 {
-    int kind;
+    int kind, n;
+    trait_object *trait;
 
-    if (!PyArg_ParseTuple(args, "i", &kind)) {
-        return -1;
+    n = PyTuple_GET_SIZE(args);
+
+    if (n == 0){
+        kind = 0;
+    }
+    else if (!PyArg_ParseTuple(args, "i", &kind)) {
+        return NULL;
     }
 
     if ((kind >= 0) && (kind <= 8)) {
+        trait = (trait_object*) PyType_GenericNew(trait_type, args, kw);
```

Comment:
Spacing consistency nit: let's preserve consistency with the existing formatting where possible
```suggestion
        trait = (trait_object *)PyType_GenericNew(trait_type, args, kw);
```

Reply:
I'm not sure whether this is the best approach to the actual allocation; am investigating what other Python C extensions do here.